tests/test_all.py

```diff
@@ -54,4 +68,14 @@ def test_fix_p_info_filesize_and_blocksize(self):
         pass
 
     def test_fix_overlay(self):
-        pass
+        with tempfile.NamedTemporaryFile() as fd:
+            urt = UpxRecoveryTool("tests/samples/overlay_8", fd.name, False)
+            urt.fix()
+
+            pre_size = os.path.getsize("tests/samples/overlay_8")
```

Comment:
What if this test gets executed before `test_get_overlay_size` ? In this case, `test_get_overlay_size` will fail because the file will have been fixed, no?

Reply:
It's a good idea to join both tests but there are 2 problems. First, the `get_overlay_size` method (tested in `test_get_overlay_size`) is executed inside the `fix` method. I'd prefer not to duplicate the content of the `fix` method inside the tests to avoid the need to remember that it should also be updated in one of the tests. The second one is that `get_overlay_size` will always return the same value as it is checking the input buffer that is read-only.

Reply:
I wasn't suggesting joining tests, my concern was that once you fix a sample file in one test, another one will start failing (because there is no more overlay), or it is not the case?

Reply:
All read files are opened in read-only mode. Fixed content is written in another different file.